cmake/kokkos_arch.cmake

```diff
@@ -845,14 +845,35 @@ IF (KOKKOS_ENABLE_OPENACC)
       Clang -Xopenmp-target=nvptx64-nvidia-cuda -march=${CLANG_CUDA_ARCH}
             -fopenmp-targets=nvptx64-nvidia-cuda
     )
+    IF(DEFINED ENV{CUDA_PATH})
+      COMPILER_SPECIFIC_LINK_OPTIONS(Clang -L$ENV{CUDA_PATH}/lib64)
+    ENDIF()
+    COMPILER_SPECIFIC_LIBS(
+      Clang -lcudart
+      NVHPC -cuda
+    )
```

Comment:
Why is `FindCUDATookit` not sufficient?

Reply:
NVHPC installs the `nvcc` compiler in two locations: one is in `${NVHPCROOT}/cuda/bin` and the other is in `${NVHPCROOT}/compilers/bin`. 
As I know, for the CUDA backend, either path will be OK, but the OpenACC Clacc compiler should use the first path.
That's why a separate `CUDA_PATH` is used, but it is still possible to use the `CUDA_ROOT` returned by `FindCUDAToolkit` for the OpenACC Clacc compiler too.

Reply:
@masterleinad Can you help on this (separating the Cuda backend from using FindCUDAToolkit)? I don't fully understand how the Kokkos CMake handles the CUDA backend.

Reply:
We can try to figure this out in a follow-up pull request.